public/js/components/Frames.js

```diff
@@ -60,40 +46,66 @@ const Frames = React.createClass({
     });
   },
 
+  renderFrame(frame) {
+    const { selectedFrame, selectFrame } = this.props;
+
+    const selectedClass = (
+      selectedFrame && (selectedFrame.id === frame.id ? "selected" : "")
+    );
+
+    return dom.li(
+      { key: frame.id,
+        className: `frame ${selectedClass}`,
+        onMouseDown: () => selectFrame(frame),
+        tabIndex: 0
+      },
+      renderFrameTitle(frame),
+      renderFrameLocation(frame)
+    );
+  },
+
+  renderFrames() {
+    let { frames } = this.props;
+
+    const numFramesToShow = this.state.showAllFrames
+      ? frames.size : NUM_FRAMES_SHOWN;
+    frames = frames.slice(0, numFramesToShow);
+
+    return dom.ul({}, frames.map(frame => this.renderFrame(frame)));
+  },
+
+  renderToggleButton() {
+    const { frames } = this.props;
+    let buttonMessage = this.state.showAllFrames
+      ? L10N.getStr("callStack.collapse") : L10N.getStr("callStack.expand");
+
+    if (frames.size < NUM_FRAMES_SHOWN) {
+      return null;
+    }
+
+    return dom.div(
+      { className: "show-more", onClick: this.toggleFramesDisplay },
+      buttonMessage
+    );
+  },
+
   render() {
-    const { frames, selectedFrame, selectFrame } = this.props;
-    const numFramesToShow = this.state.showAllFrames ? frames.length : 7;
-    let framesDisplay;
+    const { frames } = this.props;
 
     if (frames.length === 0) {
```

Comment:
`frames` appears to be changing back and forth between an Array of `length` 0 and an Immutable Set with a `size`.

That's what was initially causing #1056, because `frames.length < numFramesToShow` would never return true, because `frames.length` was always `undefined` if there were any frames. This new PR works but we've still got two different types of data coming into this component.

Still trying to figure out where that's happening. This PR works but I still feel like `frames` needs to consistently be a Seq, or either `null` or a Seq. Switching between an Array and a Seq is probably how this bug appeared in the first place.